src/components/map/RouteLayer.js

```diff
@@ -141,7 +168,9 @@ class RouteLayer extends Component {
           // ...and the last stop is first.
           const isLast = index === 0;
 
-          const hfp = this.getStopTimes(stop);
+          const hfp = selectedJourney
+            ? this.getStopTimesForJourney(stop)
+            : this.getAllStopTimes(stop);
```

Comment:
Uses separate methods depending on if a journey is selected. The stop should only get the hfp times of the selected journey. The shape of the result is the same, but if no journey is selected the result is cached.